TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java

```diff
@@ -296,16 +314,40 @@ public static void main(String[] args) throws IOException, InterruptedException,
                         Duration.ofSeconds(params.lookaheadTimeSeconds))) {
                     var tr = new TrafficReplayer(uri, buildAuthTransformerFactory(params),
                             params.allowInsecureConnections, params.numClientThreads,  params.maxConcurrentRequests);
+                    setupShutdownHookForReplayer(tr);
                     var tupleWriter = new SourceTargetCaptureTuple.TupleToFileWriter(bufferedOutputStream);
                     var timeShifter = new TimeShifter(params.speedupFactor);
                     tr.runReplayWithIOStreams(Duration.ofSeconds(params.observedPacketConnectionTimeout),
                             blockingTrafficStream, bufferedOutputStream, timeShifter, tupleWriter);
-                    log.info("reached the end of the ingestion output stream");
+                    log.info("Done processing TrafficStreams");
                 }
             }
         }
     }
 
+    private static void setupShutdownHookForReplayer(TrafficReplayer tr) {
+        var weakTrafficReplayer = new WeakReference<>(tr);
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
Neither did I, but I did learn that logging facilities use it.  It's a bummer that they are run in a nondeterministic order because our logging will just cut out (as I'm sure that you saw in the code & duplicative System.err statements).